# Project: Global Compensation Dynamics in Data Science (2020-2022)

**Author:** Saicharan K , 
**Date:** 2-1-2026 . 

## 1. Executive Summary

This project performs a longitudinal analysis of data science salaries to identify the primary drivers of compensation, The goal is to provide a data-driven framework for estimating fair market value in a volatile global market.

## 2. Notebook Structure

This analysis is divided into 7sectional components:

- **01 Context and Questions:** Problem statement, data dictionary, and strategic questions.
- **02 Data Integrity:** Cleaning, type-casting, and missing value imputation.
- **03 Distribution Science :** Univariate analysis of salaries and demographics.
- **04 Compensation Structure:** Bivariate analysis of salary drivers.
- **05 Cohort Analysis:** Temporal trends (2020 vs 2022).
- **06 Statistical validation:** Hypothesis testing (T-tests, ANOVA) to validate findings.
- **07 Bias and Limits:** Critical assessment of dataset representation.

In [41]:
# checking all libraries to negociate any issues or errors by importing 

In [42]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sn
from scipy import stats # for statistical purpose
from datetime import date

In [52]:
#loading dataset
df = pd.read_csv('../data/ds_salaries_cleaned.csv')

In [44]:
pd.set_option('display.max_columns',None)
pd.set_option('display.float_format','{:.2f}.'.format)
sn.set_style('whitegrid')
plt.rcParams['figure.figsize'] = (10,6)
print("environment setup is completed")

environment setup is completed


In [45]:
print(f"The dataset shape is  : {df.shape}")

The dataset shape is  : (607, 12)


In [46]:
display(df.head())

,Unnamed: 0,work_year,experience_level,employment_type,job_title,salary,salary_currency,salary_in_usd,employee_residence,remote_ratio,company_location,company_size
0,0,2020,MI,FT,Data Scientist,70000,EUR,79833,DE,0,DE,L
1,1,2020,SE,FT,Machine Learning Scientist,260000,USD,260000,JP,0,JP,S
2,2,2020,SE,FT,Big Data Engineer,85000,GBP,109024,GB,50,GB,M
3,3,2020,MI,FT,Product Data Analyst,20000,USD,20000,HN,0,HN,S
4,4,2020,SE,FT,Machine Learning Engineer,150000,USD,150000,US,50,US,L


In [47]:
display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 607 entries, 0 to 606
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Unnamed: 0          607 non-null    int64 
 1   work_year           607 non-null    int64 
 2   experience_level    607 non-null    object
 3   employment_type     607 non-null    object
 4   job_title           607 non-null    object
 5   salary              607 non-null    int64 
 6   salary_currency     607 non-null    object
 7   salary_in_usd       607 non-null    int64 
 8   employee_residence  607 non-null    object
 9   remote_ratio        607 non-null    int64 
 10  company_location    607 non-null    object
 11  company_size        607 non-null    object
dtypes: int64(5), object(7)
memory usage: 57.0+ KB


None

In [48]:
#empty cell

## 3. Data dictionary
The dataset contains 607 rows with the foloowing categorical definations:

## Experience Level
- **EN:** Entry-level / Junior
- **MI:** Mid-level / Intermediate
- **SE:** Senior-level / Expert
- **EX:** Executive-level / Director

### Employment Type
- **FT:** Full-time
- **PT:** Part-time
- **CT:** Contract
- **FL:** Freelance

### Remote Ratio
- **0:** On-site (No remote work)
- **50:** Hybrid (Partially remote)
- **100:** Remote (Fully remote)
- 
### Key Measures
- **salary_in_usd:** The standardized compensation metric used for all comparisons (target variable).
- **company_location:** Country code of the employer (ISO 3166-1 alpha-2 code).

In [49]:
#empty cell

## 4. Strategic Questions & Hypotheses
This analysis is guided by the following core questions:

**1. The "Seniority Premium" Analysis**
> *Hypothesis:* The salary gap between Senior (SE) and Executive (EX) roles is statistically significant, but the gap between Entry (EN) and Mid (MI) is narrowing due to demand for junior talent.

**2. The Remote Work Arbitrage**
> *Hypothesis:* Fully remote roles (100) command a higher average salary than hybrid roles (50) due to the global competition for talent, effectively decoupling salary from local cost-of-living.

**3. Geographic Dominance**
> *Hypothesis:* US-based companies pay significantly higher salaries (>$50k premium) compared to the global average, even for the same job titles.

**4. Temporal Inflation**
> *Hypothesis:* Salaries for "Data Engineers" have grown faster year-over-year (2020-2022) than "Data Scientists" due to the shifting focus towards infrastructure and MLOps.

In [ ]:
#notebook -1 end